src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspace.kt

```diff
@@ -27,6 +31,9 @@ interface CargoWorkspace {
 
     val packages: Collection<Package>
     fun findPackage(name: String): Package? = packages.find { it.name == name }
+    fun findPackageForFile(file: VirtualFile): Package?
+    /** for internal usage only */
+    fun resetPackageIndex()
```

Comment:
I don't like this method in public API but I can't come up with a better way to allow `CargoProjectsServiceImpl` call index reset